docs/arcaflow/workflows/flow-control.md

```diff
@@ -136,60 +140,74 @@ input:
           type:
             type_id: list
             items:
-              type_id: object
               id: SubRootObject
-              properties:
-                loop_id:
-                  type:
-                    type_id: integer
-                name:
-                  type:
-                    type_id: string
-
+              type_id: ref
+    SubRootObject:
+      id: SubRootObject
+      properties:
+        loop_id:
+          type:
+            type_id: integer    
+        ratio:
+          type:
+            type_id: float                                
+        name:
+          type:
+            type_id: string
+            
 steps:
   foreach_loop:
     kind: foreach
     items: !expr $.input.iterations
-    workflow: subworkflow-repeat-name.yaml
+    workflow: subworkflow.yaml
```

Comment:
Since you're passing the input straight through to the subworkflow, consider using a namespace to refer to the schema in the subworkflow instead of redefining it here.

Reply:
i think i did this